src/bluesky/tests/test_fly_plan.py

```diff
@@ -0,0 +1,85 @@
+import functools
+from collections import defaultdict
+
+import pytest
+from ophyd import Device
+from ophyd.sim import NewTrivialFlyer, StatusBase
+
+from bluesky.plans import fly
+from bluesky.tests import requires_ophyd
+
+
+def call_counter(func):
+    @functools.wraps(func)
+    def inner(self, *args, **kwargs):
+        self.call_counts[func.__name__] += 1
+        return func(self, *args, **kwargs)
```

Comment:
```suggestion
        ret = func(self, *args, **kwargs)
        self.call_counts[func.__name__] += 1
        return ret
```

Otherwise the resource path in the test class was never being hit.

Might be worth change that test to `==1` instead?

Reply:
This also suggests that there is a second bug where we accept a datum that does not yet have its matching resource?